### PR TITLE
Add PATCH /api/articles/:artcle_id model and controller, as well as 400 error handling

### PR DESCRIPTION
**Pull request by davidptclark**
Adds:
- GET /api/articles/:artcle_id model and controller
-  Tests 200 status - both positive and negative integers
- Tests 400 status - invalid id, valid id but invalid body of inc_vote object

Already existing:
- 404 (path not found) and 500 (server error) already handled from previous ticket implementation

### CODE REVIEW

**Comment by davidptclark** [comment]
P.S: Added minor commit post-pull request to correct test description.

File: __tests__/app.test.js [hunk 1]
```diff
@@ -71,4 +71,69 @@ describe('app', () => {
...
+  describe('PATCH -  /api/articles/:article_id', () => {
+    //404 and 500 already covered
+    test('Status: 200 - should respond with updated article', () => {
+      const voteUpdate = { inc_votes: 10 };
+      return request(app)
+        .patch('/api/articles/1')
+        .send(voteUpdate)
+        .expect(200)
+        .then(({ body: article }) => {
+          expect.objectContaining({
```
Comment (daviidbartlett) [inline __tests__/app.test.js:83]: are you trying to nested destructure here? You'll want to wrap article in {}

File: __tests__/app.test.js [hunk 2]
```diff
@@ -71,4 +71,69 @@ describe('app', () => {
...
+              author: expect.any(String),
+              title: expect.any(String),
+              article_id: expect.any(Number),
+              body: expect.any(String),
+              topic: expect.any(String),
+              created_at: expect.any(Number),
+              votes: expect.any(Number),
+            },
+          });
+          expect(article.article.votes).toBe(90);
```
Comment (daviidbartlett) [inline __tests__/app.test.js:115]: you're checking your response, can you check in the db is actually being updated somehow?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -71,4 +71,69 @@ describe('app', () => {
...
+      const voteUpdate = { inc_votes: 10 };
+      return request(app)
+        .patch('/api/articles/not-an-id')
+        .send(voteUpdate)
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe('bad request');
+        });
+    });
+  });
```
Comment (daviidbartlett) [inline __tests__/app.test.js:138]: missing a test here 👀 
id not found?

File: models/articles.model.js [hunk 4]
```diff
@@ -10,3 +10,17 @@ exports.fetchArticleById = (id) => {
...
+
+exports.updateVotesById = (id, votes) => {
+  return db
+    .query(
+      'UPDATE articles SET votes = votes + $2 WHERE article_id = $1 RETURNING *;',
+      [id, votes]
+    )
+    .then(({ rows: article }) => {
+      if (article.length === 0) {
+        return Promise.reject({ status: 404, msg: 'article not found' });
```
Comment (daviidbartlett) [inline models/articles.model.js:22]: you're not testing this code!!!

**Review by daviidbartlett — COMMENTED** [review]
I missing test and some untested code

**Comment by davidptclark** [comment]
I've acted on all but one of David's comments above but re: testing the updated database - the SQL query is sending back the updated data from the table, so isn't that testing that the database has been updated?

**Pull request merged**